webrtc.html

```diff
@@ -4790,23 +4770,44 @@ <h2>Methods</h2>
-              <p class="note">It is expected that a <a>user agent</a> will have
-              a small or even fixed set of values that it will accept.</p>
+
+              <p>When the method is called, the user agent MUST run the following steps:</p>
+              <ol>
+                <li>Let <var>parameters</var> be the first argument to <code>generateCertificate</code>.</li>
+                <li>Let <var>certificateExpiration</var> be the result of converting <var>parameters</var> to an <code>RTCCertificateExpiration</code> value.</li>
+                <li>If the conversion fails with an <var>error</var>, return a promise that is <a>rejected</a> with this <var>error</var>.</li>
+                <li>Let <var>keygenAlgorithm</var> be the result of converting <var>parameters</var> to an <code>AlgorithmIdentifier</code> value.</li>
+                <li>If the conversion fails with an <var>error</var>, return a promise that is <a>rejected</a> with this <var>error</var>.</li>
+                <li>Let <var>normalizedAlgorithm</var> be the result of normalizing an algorithm according to the <a data-cite="!WebCryptoAPI#algorithm-normalization">
+                  algorithm normalization process</a> [[!WebCryptoAPI]]
+                  with an operation name of <code>generateKey</code> and a [[<a data-cite= "!WebCryptoAPI#dfn-supportedAlgorithms">supportedAlgorithms</a>]]
+                  value specific to production of certificates for <code>RTCPeerConnection</code>.</li>
+                <li>If the above step produces an <var>error</var>, return a promise that is <a>rejected</a> with this <var>error</var>.</li>
+                <li>If the <var>normalizedAlgorithm</var> parameter identifies an algorithm that the <a>user agent</a> cannot or will not use to
+                  generate a certificate for <code>RTCPeerConnection</code>, return a promise that is <a>rejected</a> with a <code>DOMException</code> of type <code>NotSupportedError</code>.
+                  In particular, <var>normalizedAlgorithm</var> MUST be an asymmetric algorithm that can be used to produce a signature used to authenticate DTLS connections.</li>
+                <li>Let <var>p</var> be a new promise.
+                <li>Perform the remaining steps in parallel:
+                  <ol>
+                      <li>Perform the generate key operation specified by <var>normalizedAlgorithm</var> using <var>keygenAlgorithm</var>.</li>
+                      <!-- FIXME: What about extractable and usages parameters used in WebCrypto API spec. -->
+                      <li>Let <var>generatedKeyingMaterial</var> and <var>generatedKeyCertificate</var> be the private keying material and certificate generated by the above step.</li>
+                      <li>Let <var>certificate</var> be a new <code><a>RTCCertificate</a></code> object.</li>
+                      <li>If <var>certificateExpirations</var>'s <code>expires</code> value is undefined, set it to a default value.</li>
```

Comment:
You use dot notation below. I'm partial to it here as well: "If *certificateExpiration*.`expires` is `undefined`, ..."